src/moments.jl

```diff
@@ -43,30 +23,22 @@ function var(v::RealArray, w::AbstractWeights; mean=nothing,
     corrected = depcheck(:var, :corrected, corrected)
 
     if mean == nothing
-        varm(v, w, Statistics.mean(v, w); corrected=corrected)
+        _moment2(v, w, Statistics.mean(v, w); corrected=corrected)
     else
-        varm(v, w, mean; corrected=corrected)
+        _moment2(v, w, mean; corrected=corrected)
     end
 end
 
 ## var along dim
 
-function varm!(R::AbstractArray, A::RealArray, w::AbstractWeights, M::RealArray,
-                    dim::Int; corrected::DepBool=nothing)
-    corrected = depcheck(:varm!, :corrected, corrected)
-    rmul!(_wsum_centralize!(R, abs2, A, convert(Vector, w), M, dim, true),
-          varcorrection(w, corrected))
-end
-
 function var!(R::AbstractArray, A::RealArray, w::AbstractWeights, dims::Int;
               mean=nothing, corrected::DepBool=nothing)
     corrected = depcheck(:var!, :corrected, corrected)
 
     if mean == 0
-        varm!(R, A, w, Base.reducedim_initarray(A, dims, 0, eltype(R)), dims;
-                   corrected=corrected)
-    elseif mean == nothing
-        varm!(R, A, w, Statistics.mean(A, w, dims=dims), dims; corrected=corrected)
+        mean = Base.reducedim_initarray(A, dims, 0, eltype(R))
```

Comment:
why is this not tested. Also the same as above - why is it better than just calling the default replacement?

Reply:
Turns out it's not tested on master already. This is an undocumented feature of `var` (which calls `var!`). I wish it wasn't allowed, but I guess we have keep supporting it now (and add tests for it)...